src/halmos/sevm.py

```diff
@@ -1468,52 +1609,52 @@ def call_known(to: Address) -> None:
                     #
                     block=ex.block,
                     #
-                    calldata=calldata,
-                    callvalue=fund if op != EVM.DELEGATECALL else ex.callvalue,
-                    caller=caller if op != EVM.DELEGATECALL else ex.caller,
-                    this=to if op in [EVM.CALL, EVM.STATICCALL] else ex.this,
+                    context=CallContext(message=message, depth=ex.context.depth + 1),
+                    this=message.target,
                     #
                     pgm=ex.code[to],
                     pc=0,
                     st=State(),
                     jumpis={},
-                    output=None,
                     symbolic=ex.symbolic,
                     prank=Prank(),
                     #
                     solver=ex.solver,
                     path=ex.path,
                     alias=ex.alias,
                     #
-                    log=ex.log,
                     cnts=ex.cnts,
                     sha3s=ex.sha3s,
                     storages=ex.storages,
                     balances=ex.balances,
                     calls=ex.calls,
-                    failed=ex.failed,
-                    error=ex.error,
                 )
             )
 
             logs.extend(new_logs)
 
             # process result
-            for idx, new_ex in enumerate(new_exs):
-                opcode = new_ex.current_opcode()
-
-                # restore tx msg
-                new_ex.calldata = ex.calldata
-                new_ex.callvalue = ex.callvalue
-                new_ex.caller = ex.caller
+            for new_ex in new_exs:
+                # continue execution in the context of the parent
+                # pessimistic copy because the subcall results may diverge
+                subcall = new_ex.context
+
+                # restore context
+                new_ex.context = deepcopy(ex.context)
+                new_ex.context.trace.append(subcall)
                 new_ex.this = ex.this
 
+                if subcall.stuck():
+                    # internal errors abort the current path,
+                    # so we don't need to add it to the worklist
+                    out.append(new_ex)
+                    continue
+
```

Comment:
i think you can explicitly mark new_ex as stuck here, so that `stuck()` can be made O(1).